pkg/proxy/policy/policy.go

```diff
@@ -0,0 +1,60 @@
+// Package policy defines the network policy that the proxy can choose to enforce.
+package policy
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"strings"
+
+	"github.com/elazarl/goproxy"
+)
+
+type Policy struct {
+	Rules []Rule
+}
+
+func (p Policy) EnforcePolicy(req *http.Request, ctx *goproxy.ProxyCtx) (*http.Request, *http.Response) {
```

Comment:
nit: Since it's a method  on the Policy, maybe just "Enforce" or, better yet, "Apply"